shift_detector/checks/conditional_probabilities_check.py

```diff
@@ -20,60 +84,105 @@ class ConditionalProbabilitiesCheck(Check):
     :param min_confidence: a float between 0 and 1. This parameter impacts the size
-        of the result. The higher ``min_confidence`` the more rules are generated.
+        of the result. The higher ``min_confidence`` the more rules are considered.
     :param rule_limit: an int greater than or equal to 0. At most ``rule_limit``
-        rules are printed as a result of executing this check. The rules are sorted
-        according to their significance.
+        rules are printed in each section of the report as a result of executing this check.
+        Rules are always sorted according to their significance.
     :param min_delta_supports: a float between 0 and 1. Rules whose support
-        values exhibit an absolute difference of less than ``min_delta_supports`` are not printed.
+        values exhibit an absolute difference of less than ``min_delta_supports`` are not considered
+        to indicate a shift.
     :param min_delta_confidences: a float between 0 and 1. Rules whose confidence
-        values exhibit an absolute difference of less than ``min_delta_confidences`` are not printed.
+        values exhibit an absolute difference of less than ``min_delta_confidences`` are not considered
+        to indicate a shift.
+    :param number_of_bins: an int greater than 0. Numerical columns are binned into ``number_of_bins``
+        bins.
+    :param number_of_topics: an int greater than 0. Textual columns are embedded into ``number_of_topics``
+        topics.
     """
 
     def __init__(self, min_support=0.01, min_confidence=0.15, rule_limit=5, min_delta_supports=0.05,
-                 min_delta_confidences=0.05):
+                 min_delta_confidences=0.05, number_of_bins=50, number_of_topics=20):
         assert 0 <= min_support <= 1, 'min_support expects a float between 0 and 1'
         assert 0 <= min_confidence <= 1, 'min_confidence expects a float between 0 and 1'
         assert 0 <= rule_limit, 'rule_limit expects an int greater than or equal to 0'
         assert 0 <= min_delta_supports <= 1, 'min_delta_supports expects a float between 0 and 1'
         assert 0 <= min_delta_confidences <= 1, 'min_delta_confidences expects a float between 0 and 1'
+        assert 0 < number_of_bins, 'number_of_bins expects an int greater than 0'
+        assert 0 < number_of_topics, 'number_of_topics expects an int greater than 0'
         self.min_support = float(min_support)
         self.min_confidence = float(min_confidence)
         self.rule_limit = int(rule_limit)
         self.min_delta_supports = float(min_delta_supports)
         self.min_delta_confidences = float(min_delta_confidences)
+        self.number_of_bins = int(number_of_bins)
+        self.number_of_topics = int(number_of_topics)
 
     def run(self, store):
-        rules, examined_columns = store[
-            ConditionalProbabilitiesPrecalculation(self.min_support, self.min_confidence)]
+        pre_calculation = store[
```

Comment:
better to call it `precalculation_result`

Reply:
Good idea